### PR TITLE
Debuging-likes WIP

### PR DESCRIPTION
**Pull request by N-Patarov**

### CODE REVIEW

File: lib/bgsite_official_web/live/tag_live.ex [hunk 1]
```diff
@@ -0,0 +1,7 @@
+defmodule TagLive do
```
Comment (ykostov) [inline lib/bgsite_official_web/live/tag_live.ex:1]: Must be 
```
defmodule BgsiteOfficialWebTagLive do
  use BgsiteOfficialWeb, :live_view
```

File: lib/bgsite_official_web/live/tag_live.ex [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+  def render(assigns) do
+    render WebsitesView, "show.html", assigns
```
Comment (ykostov) [inline lib/bgsite_official_web/live/tag_live.ex:4]: Why `Websites`? Its a Tag controller, not websites'

File: lib/bgsite_official_web/router.ex [hunk 3]
```diff
@@ -79,14 +79,14 @@ defmodule BgsiteOfficialWeb.Router do
     get "/admin/register", AdminRegistrationController, :new
     post "/admin/register", AdminRegistrationController, :create
     get "/admin", AdminController, :index
+    live "/website_live/:id", WebsiteLive
```
Comment (ykostov) [inline lib/bgsite_official_web/router.ex:82]: Well, this change has no purpose, you move a live router from `:require_authenticated_admin` to `:require_authenticated_admin`, why?

File: lib/bgsite_official_web/templates/tag/show.html.leex [hunk 4]
```diff
@@ -38,8 +44,13 @@
...
 
+ <% end %>
 
  </div>
 </div>
+
+<%= def handle_event("like",_,socket) do
+    BgsiteOfficial.WebsiteLive.inc_likes(socket.assings.post)
+    {:noreply,socket}
+    end %>
```
Comment (ykostov) [inline lib/bgsite_official_web/templates/tag/show.html.leex:56]: This is totally wrong, Elixir Phoenix is not PHP we dont do that here. Its a `html.leex` template, not controller /live view/

**Review by ykostov — CHANGES_REQUESTED** [review]

File: lib/bgsite_official_web/live/tag_live.ex (hunk 1 above)
Comment (rmilushev) [reply lib/bgsite_official_web/live/tag_live.ex:1]: I guess @ykostov you mean: 
```eilixir
  defmodule BgsiteOfficialWeb.TagLive do
```
;)

File: lib/bgsite_official_web/templates/tag/show.html.leex [hunk 5]
```diff
@@ -15,37 +13,32 @@
...
          <% end %> -->
-
          <%= if @current_admin do %>
-         <h5 class="card-title text-center"><%= link website.title , to: Routes.live_path(@conn, BgsiteOfficialWeb.WebsiteLive, website) %></h5>
+          <div class="float-right">
+          <a href="#" phx_click="like" phx-target=@myself>
+            <i class="far fa-heart"></i>
+            <p class="card-text "><small class="text-muted"><%= @likes %></small></p>
+          </a>
+         </div>
```
Comment (ykostov) [inline lib/bgsite_official_web/templates/tag/show.html.leex:24]: IMO we dont need `<a href>` + `<div>`. Click event can be done on `<i class>` as well:
```
<i class="far fa-heart" phx-click="like"></i>
<p class="card-text "><small class="text-muted"><%= @likes %></small></p>
```

**Review by ykostov — COMMENTED** [review]
Great overall work :+1: Step 2 - Consider saving `<%= @likes %>` and assign in socket (on refresh likes disappear) You can refer to: https://github.com/N-Patarov/bgsite_official/blob/main/lib/bgsite_official_web/live/website_live.ex#L33-L40

File: lib/bgsite_official_web/live/websites_like.ex [hunk 6]
```diff
@@ -14,21 +15,24 @@ defmodule BgsiteOfficialWeb.WebsitesLike do
...
   end
-  # 
-  # def handle_event("add_like", %{"website-id" => website_id}, socket) do
-  #   Home.bump_site_likes(website_id)
-  #   {:noreply, socket}
-  # end
+
+  def handle_event("add_like", %{"website-id" => website_id}, socket) do
+     Home.bump_site_likes(website_id)
+     current_likes_count = website.likes
```
Comment (ykostov) [inline lib/bgsite_official_web/live/websites_like.ex:31]: there aint connection between socket assigned website and this `website.likes`, Try this
```
website = socket.assigns[:website]
```
before `current_likes_count = website.likes` :)

Comment (rmilushev) [reply lib/bgsite_official_web/live/websites_like.ex:31]: IMO you are missing to fetch the `website` based on `website_id`. Is there `website` struct on `socket`, actually? 
(fetching `website` by `id`: ```website = Home.get_websites!(website_id)```) Anoying plural again...

Comment (rmilushev) [reply lib/bgsite_official_web/live/websites_like.ex:31]: The idea was:
- to fetch the `current_website_likes` inside this bump function and use it to increase the likes with `1` here: https://github.com/N-Patarov/bgsite_official/blob/add-likes/lib/bgsite_official/home.ex#L27

Now you are requiring `likes` as an argument for the function, we just need `website` to bump its current likes count.

Comment (rmilushev) [reply lib/bgsite_official_web/live/websites_like.ex:31]: I would fix `bump..` like this:

```elixir
def bump_site_likes(%Websites{} = website) do
  updated_likes_count = website.likes + 1
  Home.update_websites(website, %{likes: updated_likes_count})
end
```

File: lib/bgsite_official/home.ex [hunk 7]
```diff
@@ -24,6 +24,17 @@ defmodule BgsiteOfficial.Home do
     end
   end
 
+  def bump_likes(%Websites{} = website, likes) do
+    website = website.id
+    query = from(w in Websites, where: w.websites_id == ^website)
```
Comment (rmilushev) [inline lib/bgsite_official/home.ex:29]: Please, change it with the generic `get_website!`, this is the cause for our next errors related to `website.id`. Here is the proper form: 
https://github.com/N-Patarov/bgsite_official/blob/add-likes/lib/bgsite_official/home.ex#L91

File: lib/bgsite_official/home.ex [hunk 8]
```diff
@@ -24,6 +24,17 @@ defmodule BgsiteOfficial.Home do
...
+    assoc = Repo.one(query)
+    if assoc == nil do
```
Comment (rmilushev) [inline lib/bgsite_official/home.ex:31]: So if we have no websites with this id, will bump the likes of `nil` ;) ?

File: lib/bgsite_official/home.ex [hunk 9]
```diff
@@ -24,6 +24,17 @@ defmodule BgsiteOfficial.Home do
...
+      %Websites{}
+      |> Websites.changeset(%{website_id: website.id, likes: likes})
```
Comment (rmilushev) [inline lib/bgsite_official/home.ex:33]: IMO: https://github.com/N-Patarov/bgsite_official/blob/add-likes/lib/bgsite_official/home.ex#L126

**Review by rmilushev — CHANGES_REQUESTED** [review]

**Pull request merged**